packages/webapp/src/util/task.ts

```diff
@@ -119,14 +118,14 @@ export const formatSoilAmendmentTaskToFormStructure = (task: DBTask | FormTask):
 
       return {
         ...formattedTaskProduct,
-        weight_unit: isWeight ? rest.weight_unit : undefined,
-        volume_unit: !isWeight ? rest.volume_unit : undefined,
+        weight_unit: isWeight ? rest.weight_unit || undefined : undefined,
+        volume_unit: !isWeight ? rest.volume_unit || undefined : undefined,
```

Comment:
If the database data is good (nulled properly), actually do we need these checks here? I missed it before, but now I'm wondering can it just be

```suggestion
        weight_unit: rest.weight_unit ?? undefined,
        volume_unit: rest.volume_unit ?? undefined,
```

and same for the other units below... or am I forgetting about a situation?

Reply:
I'm not sure why I did these only for units... I think it should be safe to go with your suggestion. Thank you!